### ISSUE TITLE
Any plans for supporting IoT Hub connection strings?

### ISSUE THREAD

**Issue by dominicbetts** [Azure/azure-event-hubs-dotnet#167]
Are you planning to add support for IoT Hub connection strings in the way that is supported in Microsft.ServiceBus.Messaging library?

It's much more convenient to use the IoT Hub connection string directly rather than assembling a Event Hub connection string from IoT Hub's Event Hub-compatible settings.

**Comment by SatyKrish** [comment]
Do you have an ETA for IOT endpoint support? It is not working with Microsoft.Azure.EventHubs v1.0.3 Nuget package.

**Comment by serkantkaraca** [comment]
Thanks for reminding this. I will bring this to table in our first client meeting and update the thread accordingly.

**Comment by sorenhoyer** [comment]
What's the status on this? Any ETA for when we can expect IOT endpoint support? :)

**Comment by dominicbetts** [comment]
Any update on this?

**Comment by massimoappi** [comment]
Please implement supports for IoT Hub default "messages/events" endpoint. I'm porting from .NET Framework 4.6.1 to .NET Core 2.0 and now I'm stucked on this! How can I connect to "messages/events" IoT Hub default EndPoint to get all "non-standard" devices messages, not responding to IoT Hub rules for routing?

**Comment by serkantkaraca** [comment]
@massimoappi, you can get the EH connection string from IOT portal. This will at least unblock you until client supports IOT connection strings.

**Comment by dominicbetts** [comment]
@massimoappi - see https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-create-through-portal#endpoints for details of how to find the Event Hub-compatible name and endpoint.

Check the Event Hubs samples for sample code: https://github.com/Azure/azure-event-hubs/tree/master/samples/DotNet/Microsoft.Azure.EventHubs

**Comment by massimoappi** [comment]
@serkantkaraca, @dominicbetts really thansk for your replies!
Now I get the default IoT Hub endpoint "messages/events" work on my console app. Coming from FF 4.6.1, the compatible Event-Hub connection string is different from the one required by .NET Standard 2.0, and you where right when saying it was on Azure portal: strings where under my eyes, but I was unable to see them! :)

@dominicbetts Regarding your post about Event Hubs sample, I'm not able to use it within IoT Hub.. it requires an Azure Storage Account I haven't in my IoT Hub.. I mean this example is for a *true* Event-Hub service on Azure, not for the Iot complatible one.. am I right? I'm using EventHubClient class instead, to receive messages on each IoT hub partitions with .CreateReceiver and implementing IPartitionReceiveHandler interface.. are there different kind of approaces for this purpose?

Thanks a lot.

**Comment by dominicbetts** [comment]
@massimoappi - the IoT Hub's Event-Hub-compatible endpoint for receiving device-to-cloud messages is fully Event Hubs compatible. For production environments, the recommended approach is to use Event Processor Host (EPH) as shown in the sample. An EPH makes your solution more robust: see https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-dotnet-framework-getstarted-receive-eph
That tutorial also shows you how to set up the storage account.

**Comment by icebat83** [comment]
Hi, any update on this? We converted our project and we found out that we cannot be backward compatible because of this missing feature. Is there a plan when it could be added and released? Thank you.

**Comment by d1mnewz** [comment]
+1 on that - we are missing that feature.

**Comment by icebat83** [comment]
Hi again. It's really blocking us at this moment. Could you please just say that it will not be supported and we will do it in different way. Or it will be supported and we will wait with our changes. Thank you.

**Comment by d1mnewz** [comment]
@icebat83 
As a workaround, I would suggest using event hub compatible connection string as @dominicbetts mentioned.

**Comment by icebat83** [comment]
@d1mnewz 
Thank you for the advice. I know about the workaround. But to support this workaround in our software we would have to update the configuration and we would also have to update configuration of all existing projects at our customers. We would like to avoid this. But if it will not be supported we will have to do it anyway.

**Comment by icebat83** [comment]
Yes, it would work for us and based on what you wrote it would do exactly what we need.

The only question is when it could be ready? You wrote in the next release. Could you please be more specific? So we can decide whether we will wait for the change.

Thank you.

**Comment by damonbarry** [comment]
It looks like @serkantkaraca submitted PR #324, which was merged and which adds plugin functionality via the `EventHubsPlugin` class. But the only customization point it provides is `EventHubsPlugin.BeforeEventSend(EventData)`. So you can manipulate data before sending an event, but I don't see a way to manipulate a connection string before connecting.

Am I missing something? It looks like this GitHub issue is still not resolved...
